packages/dds/map/src/test/mocha/map.fuzz.spec.ts

```diff
@@ -46,6 +54,15 @@ function assertMapsAreEquivalent(a: ISharedMap, b: ISharedMap): void {
 	for (const key of a.keys()) {
 		const aVal: unknown = a.get(key);
 		const bVal: unknown = b.get(key);
+		if (key === handleKey) {
+			assert.equal(typeof aVal, IFluidHandle, "aVal should be a handle");
+			assert.equal(typeof bVal, IFluidHandle, "bVal should be a handle");
+			assert.equal(
+				(aVal as IFluidHandle).get(),
+				(bVal as IFluidHandle).get(),
+				`${a.id} and ${b.id} differ at ${key}: ${aVal} vs ${bVal}`,
+			);
+		}
```

Comment:
it would be better to not have a specific handle key, so other operations, like delete also act on it. you'll also have to allow for async validators, which could be done as a seperate change. I would expect this one to look like:
```
async function assertMapsAreEquivalent(a: ISharedMap, b: ISharedMap): Promise<void> {
	assert.equal(a.size, b.size, `${a.id} and ${b.id} have different number of keys.`);
	for (const key of a.keys()) {
		const aVal: unknown = a.get(key);
		const bVal: unknown = b.get(key);
		if (isObject(aVal) && isObject(bVal)) {
			const aObj: FluidObject<IFluidHandle> | Record<string | number | symbol, unknown> =
				aVal;
			const bObj: FluidObject<IFluidHandle> | Record<string | number | symbol, unknown> =
				bVal;
			const aHandle = "IFluidHandle" in aObj ? await aObj.IFluidHandle?.get() : aObj;
			const bHandle = "IFluidHandle" in bObj ? await bObj.IFluidHandle?.get() : bObj;
			assert.equal(
				aHandle,
				bHandle,
				`${a.id} and ${b.id} differ at ${key}: ${JSON.stringify(
					aHandle,
				)} vs ${JSON.stringify(bHandle)}`,
			);
		} else {
			assert.equal(aVal, bVal, `${a.id} and ${b.id} differ at ${key}: ${aVal} vs ${bVal}`);
		}
	}
}
```

Reply:
done in #20430

Reply:
the first part of this comment still needs to be done

Reply:
should the key come from the same pool as the set operation? can that be accessed from the reducer?